.github/workflows/getdeps_linux.yml

```diff
@@ -10,122 +10,143 @@ on:
     branches:
     - main
 
+permissions:
+  contents: read  #  to fetch code (actions/checkout)
+
 jobs:
   build:
     runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
+    - name: Show disk space at start
+      run: df -h
+    - name: Free up disk space
+      run: sudo rm -rf /usr/local/lib/android
+    - name: Show disk space after freeing up
+      run: df -h
+    - name: Update system package info
+      run: sudo apt-get update
+    - name: Install system deps
+      run: sudo python3 build/fbcode_builder/getdeps.py --allow-system-packages install-system-deps --recursive fbthrift
```

Comment:
This part worries me. I don't think we want to install system thrift when building thrift.

Reply:
> This part worries me. I don't think we want to install system thrift when building thrift.

none of the getdeps manifest request a system dep for thrift, and not of their transient deps do either.  Even it if was accidentally added somehow (e.g. someone manually installs ubuntu thrift-compiler), I think still good as the topo-sorted getdeps paths are added in front of the various _PATH env vars.   

For further reassurance, not had an issue of folly system packages being found rather than the getdeps built ones

Reply:
Nevermind, I incorrectly assumed that this installs a system version of thrift itself in addition to dependencies.